common/winobjc.versioning.common.targets

```diff
@@ -0,0 +1,245 @@
+      <PackageVersion_PreReleaseFormat>$(PackageVersion_PreReleaseFormat.Replace('PackageVersion_PreReleaseLabel', '1'))</PackageVersion_PreReleaseFormat>
+
+      <PackageVersion_PreRelease Condition="'$(PackageVersion_PreRelease)' == '' And '$(PackageVersion_PreReleaseLabel)' != ''">-$([System.String]::Format('$(PackageVersion_PreReleaseFormat)', $(PackageVersion_Timestamp), $(PackageVersion_PreReleaseLabel)))</PackageVersion_PreRelease>
+
+      <PackageVersion>$(PackageVersion_Major).$(PackageVersion_Minor).$(PackageVersion_Patch)$(PackageVersion_PreRelease)</PackageVersion>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(PackageVersionOverride)' != ''">
+
+      <!-- 
+      Regex for package version override string in escaped .NET syntax:
+      ^v?(?<MAJOR>\d+)\.(?<MINOR>\d+)\.(?<PATCH>\d+)(\-(?<LABEL&>[\dA-Za-z\-\.]+))?$
+       -->
+      <_PackageVersionRegex>^v?(?&lt;MAJOR&gt;\d+)\.(?&lt;MINOR&gt;\d+)\.(?&lt;PATCH&gt;\d+)(\-(?&lt;LABEL&gt;[\dA-Za-z\-\.]+))?$</_PackageVersionRegex>
+
+      <!-- normalize version string -->
+      <PackageVersionOverride Condition="$(PackageVersionOverride.StartsWith('v'))">$(PackageVersionOverride.TrimStart('v'))</PackageVersionOverride>
+
+      <PackageVersion>$(PackageVersionOverride)</PackageVersion>
+
+      <!-- Pull out components for other targets to use -->
+      <PackageVersion_Major>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['MAJOR'].Value)</PackageVersion_Major>
+      <PackageVersion_Minor>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['MINOR'].Value)</PackageVersion_Minor>
+      <PackageVersion_Patch>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['PATCH'].Value)</PackageVersion_Patch>
+      <PackageVersion_PreReleaseLabel>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['LABEL'].Value)</PackageVersion_PreReleaseLabel>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="$([System.UInt64]::Parse($(PackageVersion_Patch))) &gt; $([MSBuild]::Subtract($([System.UInt16]::MaxValue), '1'))">
+      <PackageResourceVersion_Build>$(PackageVersion_Patch.Substring(0,4))</PackageResourceVersion_Build>
+      <PackageResourceVersion_Revision>$(PackageVersion_Patch.Substring(4))</PackageResourceVersion_Revision>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <PackageResourceVersion_Build Condition="'$(PackageResourceVersion_Build)' == ''">$(PackageVersion_Patch)</PackageResourceVersion_Build>
+      <PackageResourceVersion_Revision Condition="'$(PackageResourceVersion_Revision)' == ''">0</PackageResourceVersion_Revision>
+
+      <PackageResourceVersion>$(PackageVersion_Major).$(PackageVersion_Minor).$(PackageResourceVersion_Build).$(PackageResourceVersion_Revision)</PackageResourceVersion>
```

Comment:
Can we get a bit more explanation of the intended uses for PackageResourceVersion_Build, PackageResourceVersion_Revision, and PackageResourceVersion?